### PR TITLE
Use UDisks to find out if swap devices are encrypted

### PR DESCRIPTION
**Pull request by hughsie**
Using a heuristic is certainly not awesome.

(I'm also not sure how to test this in the self tests....)

### CODE REVIEW

File: plugins/linux-swap/fu-linux-swap.c [hunk 1]
```diff
@@ -13,46 +13,256 @@
...
+					  -1, NULL, error);
+	if (output == NULL)
+		return NULL;
+	devices = g_ptr_array_new_with_free_func (g_free);
+	g_variant_get (output, "(ao)", &iter);
+	while (g_variant_iter_next (iter, "&o", &obj))
+		g_ptr_array_add (devices, g_strdup (obj));
+
+	return g_steal_pointer (&devices);
+}
```
Comment (superm1) [inline plugins/linux-swap/fu-linux-swap.c:76]: this should go into `fu-common.c` where we have already used udisks for siimilar tasks

**Comment by superm1** [comment]
So if switching to using udisks2 for the backing, why not just have the plugin walk all the block devices udisks2 knows about and look for `http://storaged.org/doc/udisks2-api/latest/gdbus-org.freedesktop.UDisks2.Swapspace.html`?  I guess I don't know if `file` swap shows in that though would be the one potential catch.

**Comment by hughsie** [comment]
> I don't know if file swap shows in that though would be the one potential catch.

Exactly that, no swap files.

File: plugins/linux-swap/fu-linux-swap.c (hunk 1 above)
Comment (hughsie) [reply plugins/linux-swap/fu-linux-swap.c:76]: Refactoring now.

Comment (hughsie) [reply plugins/linux-swap/fu-linux-swap.c]: Okay, how about something like that.

File: plugins/linux-swap/fu-linux-swap.c [hunk 2]
```diff
@@ -13,46 +13,128 @@
...
+			if (g_strcmp0 (ty, "partition") == 0) {
+				self->enabled_cnt++;
+				if (!fu_linux_swap_verify_partition (self, fn, error))
+					return NULL;
+			} else if (g_strcmp0 (ty, "file") == 0) {
+				self->enabled_cnt++;
+				if (!fu_linux_swap_verify_file (self, fn, error))
+					return NULL;
+			} else {
+				g_warning ("unknown swap type: %s [%s]", ty, fn);
```
Comment (superm1) [inline plugins/linux-swap/fu-linux-swap.c:118]: so with this refactor I was going to suggest all the swap should be evaluted when attributes are evaluated.  Although people don't usually change it, this isn't static information.

**Comment by superm1** [comment]
New approach is great.

File: plugins/linux-swap/fu-linux-swap.c (hunk 2 above)
Comment (hughsie) [reply plugins/linux-swap/fu-linux-swap.c:118]: Agree. I think the FuLinuxSwap gets recreated whenever we change /proc/swaps, right?

Comment (superm1) [reply plugins/linux-swap/fu-linux-swap.c:118]: Oh yeah it does.. I completely forgot we have `fu_plugin_linux_swap_changed_cb ` without that in front of me.  Yes carry on.

**Review by superm1 — APPROVED** [review]

**Comment by hughsie** [comment]
@superm1 I see a Fedora crash in CI... is that a daemon segfault? Help!

**Comment by superm1** [comment]
> @superm1 I see a Fedora crash in CI... is that a daemon segfault? Help!

Looks like you got it sorted, right?

And the Debian CI failure it's an upstream TSS bug: https://github.com/tpm2-software/tpm2-tss/issues/1876

**Pull request merged**